source/braille.py

```diff
@@ -1441,11 +1441,150 @@ def _setCursor(self, info: textInfos.TextInfo):
+		if not config.conf["braille"]["showSelection"]:
+			self._currentSelection = None
+			return api.getReviewPosition().copy()
+		info: textInfos.TextInfo
+		try:
+			info = self.obj.makeTextInfo(textInfos.POSITION_SELECTION)
+		except (LookupError, RuntimeError):
+			self._currentSelection = None
+			info = api.getReviewPosition().copy()
+			# Info should be collapsed, but it is not always, at least when
+			# switching from focus mode to browse mode.
+			info.collapse()
+			return info
+		if info.isCollapsed:
+			# Cursor
+			self._currentSelection = None
+			info = api.getReviewPosition().copy()
+			# Info should be collapsed, but it is not always, at least when
+			# switching from focus mode to browse mode.
+			info.collapse()
+			return info
+		if (
+			self._currentSelection is None
+			or self._currentSelection.start != info.start
+			or self._currentSelection.end != info.end
+		):
+			# Selection changed, update also review position
+			self._currentSelection = info.copy()
+			if self.obj.isTextSelectionAnchoredAtStart:
+				self._currentSelection.move(textInfos.UNIT_CHARACTER, -1, "end")
+			self._currentSelection.collapse(end=self.obj.isTextSelectionAnchoredAtStart)
+			# Enqueue to avoid recursion error
+			queueHandler.queueFunction(queueHandler.eventQueue, api.setReviewPosition, self._currentSelection)
+			self._currentSelection = info.copy()
+			return info
+		# Selection unchanged
+		readingUnit: textInfos.TextInfo = api.getReviewPosition().copy()
+		readingUnit.expand(self._getReadingUnit())
+		if (
+			readingUnit.start >= info.end
+			or readingUnit.end <= info.start
+		):
+			# Review position is outside of selection
+			self.brailleSelectionStart = self.brailleSelectionEnd = None
+			info = api.getReviewPosition().copy()
+			# Info should be collapsed, but it is not always, at least when
+			# switching from focus mode to browse mode.
+			info.collapse()
+			return info
+		else:
+			# Review position is within selection
+			# Selection may not contain whole reading unit.
+			if readingUnit.start < self._currentSelection.start:
+				readingUnit.start = self._currentSelection.start
+			if readingUnit.end > self._currentSelection.end:
+				readingUnit.end = self._currentSelection.end
+			self._withinSelection = True
+			return readingUnit
 
-	def _getSelection(self):
-		return api.getReviewPosition().copy()
+	def update(self):
+		"""Updates this region.
+		Adds functionality to properly move within selection to the next
+		and previous line and to start and end of current line, to scroll forward
+		and back to the next and previous line, and to move to focus,
+		when selection is shown.
+		"""
+		previousReadingUnit: textInfos.TextInfo | None = getattr(self, "_readingInfo", None)
+		super().update()
+		if (
+			not config.conf["braille"]["showSelection"]
+			or not self._withinSelection
+			or self._currentScriptName is None
+			or previousReadingUnit is None
+		):
+			return
+		from globalCommands import commands

```

Comment:
Please explain what you're trying to achieve here. It concerns me that a braille region needs to know about the command that's trying to execute the update method.

Reply:
This is hack which purpose is to make it possible to use these "basic" gestures when within selection and "show selection" is enabled. As to commands they are what I have thought there should be available. I think it is important/useful that these commands would be working also when review position is within selection and "show selection" is enabled (see also BrailleHandler.scrollToCursorOrSelection).

Reply:
I agree that changing behavior depending on who called the method is concerning. Can this be moved to the scripts - if they need to cause different behavior then the logic belongs to their code.

Reply:
I think this is not a question who called method. This is case where braille is tethered to review, show selection is enabled, and reading unit contains some selected text. Purpose is to get correct part of reading unit to be displayed in braille in that special case when these scripts are executed.

I am unsure if it would make things simpler if logic would be moved to all those scripts, and honestly I do not know what logic should be in scripts. I have also understood that concern has been that this pr is complex.